### ISSUE TITLE
Energy plot issue

### ISSUE THREAD

**Issue by Edwintenhaaf** [TheFes/cheapest-energy-hours#68]
Just upgraded to v4.1.0 and it looks like the input from energy plots is not used anymore.

Start:
`    {## https://github.com/TheFes/cheapest-energy-hours ##} {% from
    "cheapest_energy_hours.jinja" import cheapest_energy_hours %} {{
    cheapest_energy_hours('sensor.nordpool_kwh_nl_eur_3_10_021', start="19:00",
    end="07:00", time_format="%Y-%m-%d %H:%M:%S", include_tomorrow=true, program
    ="wasmachine 60 graden")}}`
returns:  **2023-11-10 03:00:00**

End:
`    {## https://github.com/TheFes/cheapest-energy-hours ##} {% from
    "cheapest_energy_hours.jinja" import cheapest_energy_hours %} {{
    cheapest_energy_hours('sensor.nordpool_kwh_nl_eur_3_10_021', start="19:00",
    end="07:00", time_format="%Y-%m-%d %H:%M:%S", include_tomorrow=true, program
    ="wasmachine 60 graden", mode='end')}}`
returns: 2023-11-10 04:00:00

energy plot:
`{{ state_attr('sensor.energy_plots', 'energy_plots')['wasmachine 60 graden'].data }}`
[1, 65.75, 83.25, 82.25, 41, 2.75, 2.75, 2.75, 2.75, 4.5, 3.25, 5.25, 5, 5.25]

{{ state_attr('sensor.energy_plots', 'energy_plots')['wasmachine 60 graden'].no_weight_points}}
6

When downgrade to version 3.60 it works again.

start_date_time: >
      2023-11-10 02:50:00
  end_date_time: >
      2023-11-10 05:50:00
      
      
Did I mis some breaking changes perhaps?

**Comment by TheFes** [comment]
please send me the output of:
```jinja
{% from "cheapest_energy_hours.jinja" import cheapest_energy_hours %}
{{ cheapest_energy_hours('sensor.nordpool_kwh_nl_eur_3_10_021', start="19:00", end="07:00", time_format="%Y-%m-%d %H:%M:%S", include_tomorrow=true, program ="wasmachine 60 graden", mode="all") }}
```
and
```
{{ state_attr('sensor.energy_plots', 'energy_plots')['wasmachine 60 graden']
```

Maybe something is broken because of the updates, I didn't really test with the `program` input

Maybe wait until I release v4.2.0 and give me the output based on that version. It will probably not fix it, but it will give me more data, and I will be able to fix it based on the latest version.

**Comment by Edwintenhaaf** [comment]
Here you go:

in version 3.6.0 the output is

{"start":"2023-11-11 03:50:00","end":"2023-11-11
06:50:00","min":0.27,"max":0.274,"time_min":"2023-11-11
04:00:00","time_max":"2023-11-11
06:00:00","average":0.27217,"weighted_average":0.2703,"list":[0.271,0.27,0.27,0.27,0.27,0.27,0.27,0.273,0.273,0.273,0.273,0.273,0.273,0.274,0.274,0.274,0.274,0.274],"cheap_now":false}

{'data': [1, 65.75, 83.25, 82.25, 41, 2.75, 2.75, 2.75, 2.75, 4.5,
3.25, 5.25, 5, 5.25], 'kwh_used': 1.231, 'no_weight_points': 6,
'last_update': '2023-09-23T15:50:00+02:00', 'complete': True}

in 4.1.0 it gives:

{"start":"2023-11-11 03:50:00","end":"2023-11-11
04:50:00","min":0.27,"max":0.271,"time_min":"2023-11-11
04:00:00","time_max":"2023-11-11
03:50:00","average":0.27017,"weighted_average":0.27,"list":[0.271,0.27,0.27,0.27,0.27,0.27],"is_now":false}

{'data': [1, 65.75, 83.25, 82.25, 41, 2.75, 2.75, 2.75, 2.75, 4.5,
3.25, 5.25, 5, 5.25], 'kwh_used': 1.231, 'no_weight_points': 6,
'last_update': '2023-09-23T15:50:00+02:00', 'complete': True}

And thanks for this great macro!

**Comment by TheFes** [comment]
Fixed in v4.2.2

**Comment by Edwintenhaaf** [comment]
Thx, that was quick!

Confirmed working on 4.2.2

`{"start":"2023-11-11 03:50:00","end":"2023-11-11 06:10:00","min":0.27,"max":0.274,"time_min":"2023-11-11 04:00:00","time_max":"2023-11-11 06:00:00","average":0.27164,"weighted_average":0.2703,"list":[0.271,0.27,0.27,0.27,0.27,0.27,0.27,0.273,0.273,0.273,0.273,0.273,0.273,0.274],"is_now":false,"datapoints_per_hour":6.0,"hours":2.33,"datapoints":14}

{'data': [1, 65.75, 83.25, 82.25, 41, 2.75, 2.75, 2.75, 2.75, 4.5, 3.25, 5.25, 5, 5.25], 'kwh_used': 1.231, 'no_weight_points': 6, 'last_update': '2023-09-23T15:50:00+02:00', 'complete': True}
`
